src/components/landing/pages/register/register.tsx

```diff
@@ -0,0 +1,156 @@
+import { ShowIf } from '../../../common/show-if/show-if'
+
+enum RegisterError {
+  NONE,
+  USERNAME_EXISTING,
+  OTHER
+}
+
+export const Register: React.FC = () => {
+  const { t } = useTranslation()
+  const config = useSelector((state: ApplicationState) => state.backendConfig)
+  const user = useSelector((state: ApplicationState) => state.user)
+
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [passwordAgain, setPasswordAgain] = useState('')
+  const [error, setError] = useState(RegisterError.NONE)
+  const [ready, setReady] = useState(false)
+
+  const doAsyncRegister = async (): Promise<void> => {
+    await doInternalRegister(username, password)
+    await getAndSetUser()
+  }
+
+  const doRegisterSubmit = (event: FormEvent): void => {
+    doAsyncRegister().catch((err: Error) => {
+      if (err.message === 'Username is already existing') {
+        setError(RegisterError.USERNAME_EXISTING)
+      } else {
+        setError(RegisterError.OTHER)
+      }
+      console.error(err)
+    })
+    event.preventDefault()
+  }
+
+  useEffect(() => {
+    setReady(username !== '' && password !== '' && password.length >= 8 && password === passwordAgain)
+  }, [username, password, passwordAgain])
+
+  if (!config.allowRegister) {
+    return (
+      <Redirect to={'/login'}/>
+    )
+  }
+
+  if (user) {
+    return (
+      <Redirect to={'/intro'}/>
+    )
+  }
+
+  return (
+    <div className='my-3'>
+      <h1 className='mb-4'><Trans i18nKey='login.register.title'/></h1>
+      <Row className='h-100 d-flex justify-content-center'>
+        <Col lg={6}>
+          <Card className='bg-dark mb-4 text-start'>
+            <Card.Body>
+              <Form onSubmit={doRegisterSubmit}>
+                <Form.Group controlId='username'>
+                  <Form.Label><Trans i18nKey='login.auth.username'/></Form.Label>
+                  <Form.Control
+                    type='text'
+                    size='sm'
+                    value={username}
+                    isValid={username !== ''}
+                    onChange={(event) => setUsername(event.target.value)}
+                    placeholder={t('login.auth.username')}
+                    className='bg-dark text-white'
+                    autoComplete='username'
+                    autoFocus={true}
+                    required
+                  />
+                  <Form.Text><Trans i18nKey='login.register.usernameInfo'/></Form.Text>
+                </Form.Group>
+                <Form.Group controlId='password'>
+                  <Form.Label><Trans i18nKey='login.auth.password'/></Form.Label>
+                  <Form.Control
+                    type='password'
+                    size='sm'
+                    value={password}
+                    isValid={password !== '' && password.length >= 8}
+                    onChange={(event) => setPassword(event.target.value)}
```

Comment:
We tried to avoid putting the clear text password in the JS context via state, but I just looked the old via-email and via-ldap code up and we didn't do that there too? Do you know when that changed @mrdrogdrog